### DISCUSSION TITLE
Should the core module be bundled in all packages?

### DISCUSSION THREAD

**Discussion by Vorobey31415** [ueberdosis/tiptap#5660] [General Questions] [answered]
Hi there! 

I’m part of the WebStorm core team, and we’ve noticed that our users are facing performance issues related to the structure of your packages. Specifically, as I can see the **core** package is included in the **dist** for every Tiptap package, which leads to slower performance during development and increases the package size. This case helped us to see an interesting point and potential opportunity for performance optimization, so thank you.

Certainly, the fix that will optimize the handling of such cases already in the progress. 

But could you please clarify why the core package is bundled this way, and if there are any plans to optimize this setup in the future?

Thank you for your time, and I look forward to your response.

Best regards,
Andrei Vorobev

**Comment by nperez0111** [comment]
Hi @Vorobey31415 , when you say that the *core* package is included in our *dist* directory, do you mean the Typescript types? 
<img width="422" alt="image" src="https://github.com/user-attachments/assets/9c2b87a7-6f80-4bb6-b5a2-a58f17eee525">


I actually wasn't aware of it, and we can definitely do better!

**Reply by Vorobey31415** [reply]
Yes, sure! If this is not the intended architectural solution, then yes, it would be better to optimize. Thank you for your attention!

**Reply by hakimio** [reply]
Should there be an open issue for this?

**Reply by Vorobey31415** [reply]
I guess it's up to you, but it's better to open one just so you don't forget

**Reply by nperez0111** [reply]
Made a PR here: https://github.com/ueberdosis/tiptap/pull/5665

**Reply by hakimio** [reply]
@nperez0111 thank you for opening the PR so fast 🙂

**Reply by nperez0111** [reply, marked as answer]
This has been released with tiptap v2.8.0

**Reply by hakimio** [reply]
Awesome. Thank you 🙂
